### PR TITLE
MLE-15255 : Run the autogeneration of new Optic Code for Node Client

### PR DESCRIPTION
**Pull request by anu3990**
(This generation was conducted after replacing hints file with the one in develop branch.)

### CODE REVIEW

File: lib/plan-builder-generated.js [hunk 1]
```diff
@@ -1155,10 +1155,9 @@ periodRangeQuery(...args) {
 point(...args) {
     const namer = bldrbase.getNamer(args, 'latitude');
     const paramdefs = [['latitude', [types.XsDouble, PlanColumn, PlanParam], true, false], ['longitude', [types.XsDouble, PlanColumn, PlanParam], true, false]];
-    // Changing minArity to 1 manually since cts.point now accepts string input (MarkLogic 11.1.0 or higher needed).
```
Comment (rjrudin) [inline lib/plan-builder-generated.js:1158]: Still need to undo these changes to cts.point and cts.polygon.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: test-basic/plan-builder-generated.js [hunk 2]
```diff
@@ -1732,6 +1744,36 @@ describe('plan builder', function() {
             done();
         }).catch(done);
     }); 
+    it('vec.base64Decode#1', function(done) {
+        if(serverConfiguration.serverVersion < 12) {
+            this.skip();
+        }
+        testPlan([p.xs.string("abc")], p.vec.base64Decode(p.col("1")))
```
Comment (rjrudin) [inline test-basic/plan-builder-generated.js:1751]: What is this test actually doing? It looks like it's making sure that the response has "null" in it somewhere - how do we know that's ensuring that base64Decode works? 

I think the tests that @BillFarber  added for the vector operations for the Java Client would be much better - i.e. create a plan that binds new columns with the output of a bunch of vector operations, and verify that the bound columns have the expected values.

File: test-basic/plan-builder-generated.js [hunk 3]
```diff
@@ -2353,8 +2395,12 @@ describe('plan builder', function() {
     }); 
     it('xs.negativeInteger#1', function(done) {
         testPlan([p.xs.double(-1)], p.xs.negativeInteger(p.col("1")))
-          .then(function(response) { 
-            should(String(getResult(response).value).replace(/^ /, '')).equal('-1');
+          .then(function(response) {
+              if(serverConfiguration.serverVersion < 12) {
+                  should(String(getResult(response).value).replace(/^ /, '')).equal('-1');
+              } else {
+                  should(String(getResult(response).value).replace(/^ /, '')).equal(null);
```
Comment (rjrudin) [inline test-basic/plan-builder-generated.js:2402]: Why is there here for MarkLogic 12? Is it expected that this now returns null instead of -1?

File: test-basic/plan-builder-generated.js (hunk 2 above)
Comment (anu3990) [reply test-basic/plan-builder-generated.js:1751]: This was auto-generated. I added the skip block since it fails on server versions less than 12.

File: test-basic/plan-builder-generated.js (hunk 3 above)
Comment (anu3990) [reply test-basic/plan-builder-generated.js:2402]: Since it was auto-generated I would say yes, but for server versions less than 12, it gives -1 so added the if else block.

**Review by rjrudin — APPROVED** [review]
Approving in anticipation of manually-written tests being added for the vector operations.

**Review by llinggit — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]
